src/EnergyPlus/HeatBalFiniteDiffManager.cc

```diff
@@ -859,15 +862,22 @@ namespace HeatBalFiniteDiffManager {
             SurfaceFD(Surf).PhaseChangeTemperatureReverse = 50;
             SurfaceFD(Surf).condNodeReport = 0.0;
             SurfaceFD(Surf).specHeatNodeReport = 0.0;
+            SurfaceFD(Surf).heatSourceFluxLayerReport = 0.0;
+            SurfaceFD(Surf).heatSourceFluxEnergyLayerReport = 0.0;
 
             // Setup EMS data
-            for (int lay = 1; lay <= state.dataConstruction->Construct(ConstrNum).TotLayers; ++lay) {
+            for (int lay = 1; lay <= TotLayers; ++lay) {
                 // Setup material layer names actuators
                 int matLay = state.dataConstruction->Construct(ConstrNum).LayerPoint(lay);
                 // Actuator name format: "{SurfName}:{MaterialLayerName}"
                 std::string actName = fmt::format("{}:{}", state.dataSurface->Surface(Surf).Name, state.dataMaterial->Material(matLay).Name);
```

Comment:
All of this EMS setup here and below should be inside an  `if (state.dataGlobal->AnyEnergyManagementSystemInModel)` (well, as much of it as possible).